### PR TITLE
JAMES-2414 Run SpamAssassin ham/spam feedback in an async fashion

### PR DESCRIPTION
**Pull request by chibenwa**
In test we should be able to enable it synchronously.

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Tests run: 66, Failures: 1, Errors: 0, Skipped: 7, Time elapsed: 50.157 sec <<< FAILURE! - in org.apache.james.queue.activemq.ActiveMQMailQueueBlobTest
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] concurrentEnqueueShouldNotAlterBrowsing()  Time elapsed: 0.029 sec  <<< FAILURE!
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] java.lang.AssertionError: 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Actual and expected should have same size but actual size was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <3>
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] while expected size was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <2>
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Actual was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <["name1", "name2", "name3"]>
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Expected was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <["name1", "name2"]>
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] 	at org.apache.james.queue.api.ManageableMailQueueContract.concurrentEnqueueShouldNotAlterBrowsing(ManageableMailQueueContract.java:245)
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Results :
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Failed tests: 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   ManageableMailQueueContract.concurrentEnqueueShouldNotAlterBrowsing:245 
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Actual and expected should have same size but actual size was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <3>
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] while expected size was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <2>
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Actual was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <["name1", "name2", "name3"]>
[4d5eec9761f4356c9bcc2005f38cfd90299b5714] Expected was:
[4d5eec9761f4356c9bcc2005f38cfd90299b5714]   <["name1", "name2"]>
```

test this please

**Comment by aduprat** [comment]
It could be great to add some documentation about this configuration, but I agree to do it in another issue (maybe at the end of the sprint).

**Comment by chibenwa** [comment]
```
[50d125fe075b260abcb200de8fd6a8219991055b] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project cassandra-jmap-integration-testing: ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[50d125fe075b260abcb200de8fd6a8219991055b] [ERROR] Command was /bin/sh -c cd /james-project/server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -Xms512m -Xmx1024m -jar /james-project/server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/target/surefire/surefirebooter1834007122881301611.jar /james-project/server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/target/surefire/surefire4599002718683949244tmp /james-project/server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/target/surefire/surefire_8573914339195654809531tmp
```

test this please

**Comment by chibenwa** [comment]
```
[50d125fe075b260abcb200de8fd6a8219991055b] + docker cp keystore:/keys/keystore destination/conf
```

:-(

test this please

**Comment by chibenwa** [comment]
IMO can be considered :green_apple:

**Comment by chibenwa** [comment]
Obsolete see https://github.com/linagora/james-project/pull/1462

I keep it open for now

File: mailbox/plugin/spamassassin/src/test/java/org/apache/james/mailbox/spamassassin/SpamAssassinConfigurationTest.java [hunk 1]
```diff
@@ -20,15 +20,15 @@
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply mailbox/plugin/spamassassin/src/test/java/org/apache/james/mailbox/spamassassin/SpamAssassinConfigurationTest.java]: The configuration do not do much.

I'm not fan of boilerplate tests.

Most of the work (and tests) are handled by the loader.

**Pull request closed without merging**